src/containers/other/NFTView/NFTView.tsx

```diff
@@ -77,40 +83,107 @@ const NFTView: React.FC<NFTViewProps> = ({ onTransfer }) => {
+		const nftMoreOptions = isOwnedByYou
+			? getActionFeatures(Boolean(buyNowPrice), allBids?.length !== 0)
+			: [];
+
+		return {
+			isBiddable,
+			isOwnedByYou,
+			assetUrl,
+			nftMoreOptions,
+		};
+	}, [znsDomain, domainMetadata, account, buyNowPrice, allBids]);
+
+	// Convert highest bid as wei
+	const highestBidAsWei =
+		highestBid && ethers.utils.parseEther(highestBid.toString()).toString();
 
 	///////////////
 	// Functions //
 	///////////////
 
-	//- Callback functions
-	const onOpenBidOverlay = useCallback(() => {
-		if (!znsDomain || isOwnedByYou || !active) {
-			return;
-		}
-
-		setIsBidOverlayOpen(true);
-	}, [znsDomain, active, isOwnedByYou]);
+	const handleOnAccept = () => {
+		refetch();
+		closeModal();
+	};
 
-	// Close Make A Bid
-	const onCloseBidOverlay = () => setIsBidOverlayOpen(false);
-	// Close Domain Settings
-	const onCloseDomainSettingsOverlay = () => setIsDomainSettingsOpen(false);
 	const onBid = useCallback(async () => {
 		getPriceData();
 		getHistory();
-		onCloseBidOverlay();
 	}, [getHistory, getPriceData]);
 
+	const handleOnBid = () => {
+		onBid();
+		closeModal();
+	};
+
+	// Open Make A Bid Modal
+	const openMakeABid = () => {
+		openModal({
+			modalType: NFTViewModalType.MAKE_A_BID,
+			contentProps: {
+				domain: znsDomain!,
+				onBid: handleOnBid,
+				onClose: closeModal,
+			},
+		});
+	};
+
+	// Open Domain Settings Modal
+	const openDomainSettings = () => {
+		openModal({
+			modalType: NFTViewModalType.DOMAIN_SETTINGS,
+			contentProps: {
+				domainId: domainId,
+				onClose: closeModal,
+			},
+		});
+	};
+
+	// Open Domain Settings Modal
+	const openSetBuyNow = () => {
+		openModal({
+			modalType: NFTViewModalType.SET_BUY_NOW,
+			contentProps: {
+				domainId: domainId,
+				onCancel: closeModal,
+				onSuccess: refetch,
+			},
+		});
+	};
+
+	// Open Bid List Modal
+	const openBidList = () => {
+		openModal({
+			modalType: NFTViewModalType.BID_LIST,
+			contentProps: {
+				bids: allBids ?? [],
+				domain: viewBidsDomainData,
+				domainMetadata: domainMetadata ?? undefined,
+				onAccept: handleOnAccept,
+				wildPriceUsd,
+				isLoading: isBidDataLoading,
+				highestBid: String(highestBidAsWei),
+			},
+		});
+	};
+
 	// Dropdown Option Select
-	const onSelectNFTMoreOption = (option: Option) => {
-		if (option.title === NFT_MORE_ACTIONS_TITLE.MY_DOMAIN_SETTINGS) {
-			setIsDomainSettingsOpen(true);
-		} else if (option.title === NFT_MORE_ACTIONS_TITLE.TRANSFER_OWNERSHIP) {
-			onTransfer();
+	const onSelectOption = (option: Option) => {
+		switch (option.title) {
+			case NFT_MORE_ACTIONS_TITLE.MY_DOMAIN_SETTINGS:
+				return openDomainSettings();
+			case NFT_MORE_ACTIONS_TITLE.TRANSFER_OWNERSHIP:
+				return onTransfer();
+			case NFT_MORE_ACTIONS_TITLE.SET_BUY_NOW:
+				return openSetBuyNow();
+			case NFT_MORE_ACTIONS_TITLE.EDIT_BUY_NOW:
+				return openSetBuyNow();
+			case NFT_MORE_ACTIONS_TITLE.VIEW_BIDS:
+				return openBidList();
+			default:
+				return '';
 		}
```

Comment:
I think `onSelectOption: (option: Option) => void;` so we don't need to return for `default` case.